Fix E2E tests in non-Vagrant environment

mbarnes opened the pull request:
This 1st commit gets the E2E tests partially working in a non-Vagrant environment.

All tests pass when using Vagrant, 10 tests fail when not.  I'm hoping to fix these without touching service code.
```
Failing scenarios:
  features/cluster/deploy.feature:28  Initiate tree deployment with authentication
  features/cluster/hosts.feature:79  Verifying a cluster after adding a host (1)
  features/cluster/hosts.feature:186  Deleting a host also removes it from its cluster (1)
  features/cluster/hosts.feature:199  Deleting a host also removes it from its cluster (2)
  features/cluster/hosts.feature:212  Deleting a host also removes it from its cluster (3)
  features/cluster/list.feature:26  List all clusters with authentication
  features/cluster/restart.feature:28  Initiate cluster restart with authentication
  features/cluster/upgrade.feature:28  Initiate cluster upgrade with authentication
  features/hosts/retrieve.feature:27  Retrieve existing host with authentication
  features/hosts/retrieve.feature:49  Retrieve existing host with authentication

6 features passed, 6 failed, 0 skipped
43 scenarios passed, 10 failed, 5 skipped
280 steps passed, 10 failed, 56 skipped, 0 undefined
Took 0m55.263s
```

mbarnes committed:
e2e: Specify config file for each backend service.

ashcrow commented on features/environment.py:
```diff
@@ -342,22 +335,34 @@ def before_all(context):
             context.PROCESSES['commissaire-storage-service'] = try_start(
                 start_commissaire_service, 'commissaire-storage-service',
-                context, ['commissaire-storage-service'])
+                context, [
```
I think this service will need to copy the default and update the storage handler to point to the generated etcd port.

mbarnes replied:
Yeah you're right.  Think I have a solution.

mbarnes committed:
config: Allow reading configuration from standard input.

As with the fileinput module, specifying --config-file=- will read JSON
configuration from standard input.

mbarnes committed:
e2e: Remove unused 'times' param from try_start().

mbarnes committed:
e2e: Generate config for commissaire-storage-service.

Since the etcd service may use a random port, a static storage.conf file
won't work.  Substitute the etcd server URL in a configuration string
and feed it to commissaire-storage-service through stdin.

mbarnes committed:
e2e: Tag clusterexec test cases as @slow.

These will be skipped when not using the Vagrant environment.

mbarnes commented:
@ashcrow :arrow_up:  All tests passing for me.

Now, any ideas how we might tie this into Travis?  The test environment requires multiple git repos, if only for the config files in `commissaire-http`.

ashcrow commented:
I'll take a look tomorrow when I have time, but let's tie it in via a different PR.

ashcrow commented:
testing now

ashcrow commented on features/environment.py:
```diff
@@ -340,24 +346,41 @@ def before_all(context):
-                context, ['commissaire-storage-service'])
+                context, [
+                    'commissaire-storage-service',
+                    '--config-file', '-'],
```
Ha! Good idea!

ashcrow commented:
## Default

```
$ ./tools/behave
...
12 features passed, 0 failed, 0 skipped
50 scenarios passed, 0 failed, 8 skipped
290 steps passed, 0 failed, 56 skipped, 0 undefined
Took 0m24.318s
```

## With Vagrant
```
$ ./tools/vagrantup
...
$ ./tools/behave -D use-vagrant
...
12 features passed, 0 failed, 0 skipped
58 scenarios passed, 0 failed, 0 skipped
346 steps passed, 0 failed, 0 skipped, 0 undefined
Took 1m1.437s
```
